Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -150,14 +149,8 @@ public bool ContainsKey(string key)
 		{
 			get
 			{
-				if (_innerDictionary.ContainsKey(index))
-					return _innerDictionary[index];
-				if (_mergedInstance != null && _mergedInstance.ContainsKey(index))
-					return _mergedInstance[index];
-				if (MergedDictionaries != null)
-					foreach (var dict in MergedDictionaries.Reverse())
-						if (dict.ContainsKey(index))
-							return dict[index];
+				object value;
+				if (TryGetValue(index, out value)) return value;
```

Comment:
it make sense to only invoke `ContainsKey` to sub directories in `ContainsKey`, and not use `TryGetValue()`

Reply:
Seems, that there is bug at current master branch.
I don't know why but current implementation of ContainsKey of ResourceDictionary look follow
```javascript
		public bool ContainsKey(string key)
		{
			return _innerDictionary.ContainsKey(key);
		}
```
It is not include merged resources and when we try use ContainKey into indexer we lost recursive merged dictionaries [found values on the first surface level of MergedDictionaries collection].
```javascript
                if (_innerDictionary.ContainsKey(index))
                    return _innerDictionary[index];
                if (_mergedInstance != null && _mergedInstance.ContainsKey(index))
                    return _mergedInstance[index];
                if (_mergedDictionaries != null)
                    foreach (var dict in _mergedDictionaries.Reverse())
                        if (dict.ContainsKey(index))
                            return dict[index];
```
It possible check with new test method [fails on the current master branch]
```javascript
		[Test]
		public void MergedDictionariesDeepSublevels()
		{
			var rd0 = new ResourceDictionary();
			var rd1 = new ResourceDictionary();
			var rd2 = new ResourceDictionary { { "Key2", "Level2" } };
			rd1.MergedDictionaries.Add(rd2);
			rd0.MergedDictionaries.Add(rd1);

			Assert.That(rd0["Key2"], Is.EqualTo("Level2"));
		}
```
Also TryGetValue more fast than construction 'if (dic.ContainsKey(key)) dic[key]' and preffered to use in the current situation. As bonus, allow use C# 7 function code style.

Reply:
that's not a bug. it's on purpose

Reply:
Can you explain why?

How can I use hierarchically merged dictionaries like on WPF or UWP ?